Mathlib/Algebra/Divisibility/Basic.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2014 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Leonardo de Moura, Floris van Doorn, Amelia Livingston, Yury Kudryashov,
+Neil Strickland, Aaron Anderson
+-/
+import Mathlib.Algebra.Hom.Group
+
+/-!
+# Divisibility
+
+This file defines the basics of the divisibility relation in the context of `(comm_)` `monoid`s.
+
+## Main definitions
+
+ * `monoid.has_dvd`
+
+## Implementation notes
+
+The divisibility relation is defined for all monoids, and as such, depends on the order of
+  multiplication if the monoid is not commutative. There are two possible conventions for
+  divisibility in the noncommutative context, and this relation follows the convention for ordinals,
+  so `a | b` is defined as `∃ c, b = a * c`.
+
+## Tags
+
+divisibility, divides
+-/
+
+
+variable {α : Type _}
+
+section Semigroup
+
+variable [Semigroup α] {a b c : α}
+
+/-- There are two possible conventions for divisibility, which coincide in a `comm_monoid`.
+    This matches the convention for ordinals. -/
+instance (priority := 100) semigroupHasDvd : Dvd α :=
+  Dvd.mk fun a b => ∃ c, b = a * c
+#align semigroup_has_dvd semigroupHasDvd
+
+-- TODO: this used to not have `c` explicit, but that seems to be important
+--       for use with tactics, similar to `exists.intro`
```

Comment:
Please use `Porting note` (possibly in addition to `TODO` if it needs to be fixed before merging) so we can easily identify these.

Reply:
This TODO was actually already in the Lean 3 file. Should I mark it as such or leave it as is?